### PR TITLE
Enhancement: Allow to enable/disable PHPStan via projects config

### PR DESCRIPTION
**Pull request by OskarStark**
Follows https://github.com/sonata-project/SonataDoctrineORMAdminBundle/pull/1128

### CODE REVIEW

File: templates/project/.github/workflows/qa.yaml.twig [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+
+    steps:
+      - uses: actions/checkout@v2
+
+      - name: PHPStan
+        uses: docker://oskarstark/phpstan-ga
+        env:
+          REQUIRE_DEV: true
+        with:
+          args: analyse -c .phpstan/phpstan.neon.dist
```
Comment (jordisala1991) [inline templates/project/.github/workflows/qa.yaml.twig:29]: The file should not be in a subfolder

File: templates/project/Makefile.twig [hunk 2]
```diff
@@ -67,3 +67,9 @@ docs:
 	cd {{ branch.docsPath.toString }} && sphinx-build -W -b html -d _build/doctrees . _build/html
 .PHONY: docs
 {% endif %}
+{% if project.usesPHPStan %}
+
+phpstan:
+    docker run --env REQUIRE_DEV=true --rm -it -w=/app -v ${PWD}:/app oskarstark/phpstan-ga:latest -c .phpstan/phpstan.neon.dist
```
Comment (OskarStark) [inline templates/project/Makefile.twig:73]: @VincentLanglet WDYT? This is how I execute it in my projects

File: templates/project/.github/workflows/qa.yaml.twig (hunk 1 above)
Comment (OskarStark) [reply templates/project/.github/workflows/qa.yaml.twig:29]: I used the one from the initial PR from @core23 

@core23 shall we change that?

**Comment by OskarStark** [comment]
I executed it against `admin-bundle` and saw we are using PHPStan and Psalm, is this needed?

```diff
diff --git a/.github/workflows/qa.yaml b/.github/workflows/qa.yaml
index 01f5a6eb..e249d41b 100644
--- a/.github/workflows/qa.yaml
+++ b/.github/workflows/qa.yaml
@@ -1,22 +1,28 @@
-on: [push, pull_request]
+# DO NOT EDIT THIS FILE!
+#
+# It's auto-generated by sonata-project/dev-kit package.
+
 name: Quality assurance
+
+on:
+  push:
+    branches:
+      - master
+      - 3.x
+  pull_request:
+
 jobs:
   phpstan:
     name: PHPStan
+
     runs-on: ubuntu-latest
+
     steps:
-      - uses: actions/checkout@master
+      - uses: actions/checkout@v2
+
       - name: PHPStan
-        uses: "docker://oskarstark/phpstan-ga"
+        uses: docker://oskarstark/phpstan-ga
         env:
           REQUIRE_DEV: true
         with:
           args: analyse -c .phpstan/phpstan.neon.dist
-  psalm:
-    name: Psalm
-    runs-on: ubuntu-latest
-    steps:
-      - name: Checkout code
-        uses: actions/checkout@v2
-      - name: Psalm
-        uses: docker://vimeo/psalm-github-actions
```

File: templates/project/.github/workflows/qa.yaml.twig (hunk 1 above)
Comment (core23) [reply templates/project/.github/workflows/qa.yaml.twig:29]: I copy & pasted it from the admin bundle, which provides some custom stubs: https://github.com/sonata-project/SonataAdminBundle/tree/3.x/.phpstan

**Comment by core23** [comment]
Can you also add psalm ?

File: templates/project/.github/workflows/qa.yaml.twig (hunk 1 above)
Comment (jordisala1991) [reply templates/project/.github/workflows/qa.yaml.twig:29]: Not all the bundles / branches uses same structure

Comment (jordisala1991) [reply templates/project/.github/workflows/qa.yaml.twig:29]: And we just merge this: https://github.com/sonata-project/dev-kit/blob/49ca452d26bb3b93d6a41b5e95b1a6419214dc79/templates/project/.gitattributes.twig#L11

Comment (core23) [reply templates/project/.github/workflows/qa.yaml.twig:29]: Maybe we can find a global structure where we store all our tool configs? E.g. `/tools/$TOOL/.config`

Comment (OskarStark) [reply templates/project/.github/workflows/qa.yaml.twig:30]: If this is our current structure, lets keep it, lets run dev-kit, and improve it if needed

Comment (VincentLanglet) [reply templates/project/.github/workflows/qa.yaml.twig:33]: It's not the current structure for SonataAdmin master branch.
And it's not the same structure for psalm.

IMHO we should have consistency and move the phpstan.neon.dist to the root folder.

File: templates/project/Makefile.twig [hunk 3]
```diff
@@ -67,3 +67,9 @@ docs:
 	cd {{ branch.docsPath.toString }} && sphinx-build -W -b html -d _build/doctrees . _build/html
 .PHONY: docs
 {% endif %}
+{% if project.usesPHPStan %}
+
+phpstan:
+    docker run --env REQUIRE_DEV=true {% if project.repository.name == 'SonataDoctrineMongoDBAdminBundle' %}--env CHECK_PLATFORM_REQUIREMENTS=false {% endif %}--rm -it -w=/app -v ${PWD}:/app oskarstark/phpstan-ga:latest -c .phpstan/phpstan.neon.dist
```
Comment (VincentLanglet) [inline templates/project/Makefile.twig:73]: Can we fix the platform requirement for SonataDoctrineMongoDBAdminBundle ?

File: templates/project/.github/workflows/qa.yaml.twig (hunk 1 above)
Comment (jordisala1991) [reply templates/project/.github/workflows/qa.yaml.twig:33]: Not only for admin, the mongodb bundle does have the file on the root folder

**Pull request merged**